### PR TITLE
[image_picker] request camera permission if need

### PR DESCRIPTION
**Pull request by pepegich**
## Description

Camera permission need request if it present in manifest, because for M or great for take Photo ar Video by intent need it permission,even if the camera permission is not used.
Camera permission may be used in another package, as example flutter_barcode_reader.

## Related Issues

https://github.com/flutter/flutter/issues/29837
https://github.com/flutter/flutter/issues/29662
https://github.com/flutter/flutter/issues/29422
https://github.com/flutter/flutter/issues/29003

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerUtils.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   *
+   * <p>Camera permission may be used in another package, as example flutter_barcode_reader.
+   * https://github.com/flutter/flutter/issues/29837
+   *
+   * @return returns true, if need request camera permission, otherwise false
+   */
+  static boolean needRequestCameraPermission(Context context) {
+    boolean greatOrEqualM = Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
+    return greatOrEqualM
+        && ImagePickerUtils.isPermissionPresentInManifest(context, Manifest.permission.CAMERA);
```
Comment (cyanglaz) [inline packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerUtils.java]:
```suggestion
        && isPermissionPresentInManifest(context, Manifest.permission.CAMERA);
```

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java [hunk 2]
```diff
@@ -290,9 +306,20 @@ public void takeImageWithCamera(MethodCall methodCall, MethodChannel.Result resu
...
+      permissionManager.askForPermission(
+          Manifest.permission.CAMERA, REQUEST_CAMERA_IMAGE_PERMISSION);
+      return;
+    }
+
     launchTakeImageWithCameraIntent();
   }
 
+  private boolean needRequestCameraPermission() {
+    return permissionManager.needRequestCameraPermission();
```
[the remark this answers is not in the document]
Comment (pepegich) [reply packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java]: done

**Comment by cyanglaz** [comment]
Thank you for doing this. Before I can merge this, please rebase and also do necessary changes to make the CI happy.

File: packages/image_picker/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+## 0.5.4+2
```
Comment (cyanglaz) [inline packages/image_picker/CHANGELOG.md]:
```suggestion
## 0.5.4+2

```

**Comment by rcagantas** [comment]
Apologies if this is a noob question but is there a way to know when this lands on the beta channel?

**Comment by cyanglaz** [comment]
> Apologies if this is a noob question but is there a way to know when this lands on the beta channel?

Good question! I don't think we have a beta channel for plugins. You can use the newest version of this plugin in your code now and it would have this fix. Or you can use a specific version of the plugin. Or you can fork the plugin repo and select a version that you want to use.
If you want to know the beta channel of the `Flutter framework`. You can find details here https://github.com/flutter/flutter/wiki/Flutter-build-release-channels

**Comment by rcagantas** [comment]
Oh right. I was confused. I thought this fix was in the Flutter plugin framework and not in the plugin itself. Thanks!

**Pull request merged**